redisson/src/main/java/org/redisson/client/handler/CommandPubSubDecoder.java

```diff
@@ -47,7 +47,8 @@ public class CommandPubSubDecoder extends CommandDecoder {
 
     private static final Set<String> UNSUBSCRIBE_COMMANDS = new HashSet<>(Arrays.asList(RedisCommands.PUNSUBSCRIBE.getName(), RedisCommands.UNSUBSCRIBE.getName(), RedisCommands.SUNSUBSCRIBE.getName()));
     private static final Set<String> SUBSCRIBE_COMMANDS = new HashSet<>(Arrays.asList(RedisCommands.PSUBSCRIBE.getName(), RedisCommands.SUBSCRIBE.getName(), RedisCommands.SSUBSCRIBE.getName()));
-    private static final Set<String> MESSAGES = new HashSet<String>(Arrays.asList("subscribe", "psubscribe", "punsubscribe", "unsubscribe", "ssubscribe", "sunsubscribe"));
+    private static final Set<String> MESSAGES = new HashSet<>(Arrays.asList("subscribe", "psubscribe", "punsubscribe", "unsubscribe", "ssubscribe", "sunsubscribe"));
```

Comment:
Add test reproducing the issue

Reply:
some test added

Reply:
Sorry, I meant integration tests as I wrote in RedissonShardedTopicTest

Reply:
Sorry, I don't quite understand.
Do you mean fix RedissonShardedTopicTest because it's not working correctly, or what test do I need to write?

Reply:
Test fixing will be enough.

Reply:
Okay, now it can be merged?

Reply:
Sorry, I don't see the additional logic in the RedissonShardedTopicTest which reproduces the issue. Test logic remains the same.

Reply:
Yes, there is no additional logic. Without my changes in CommandPubSubDecoder.java, RedissonShardedTopicTest failed.

Reply:
It passes for me as well.

![image](https://user-images.githubusercontent.com/1104661/208886664-d672fd16-ca92-4f19-8745-af0c3a69ba93.png)

Reply:
It's so strange.
When I run a test it failed
![Screenshot_20221221_135237](https://user-images.githubusercontent.com/6016450/208888818-7cc297c4-46d7-4b2b-adeb-a410b6469ce9.png)

OS: Linux(Manjaro)
Kernel: 6.0.14-1-MANJARO
Redis server v=7.0.7 sha=00000000:0 malloc=jemalloc-5.3.0 bits=64 build=a077dda077766a30

Reply:
Ah, I see now. I used Redis 7.0.0 for testing. Behaviour has changed since Redis 7.0.1 

* Protocol: Sharded pubsub publish emits `smessage` instead of `message` (#10792)

They fixed that.

Reply:
Great!  Now it can be merged? :smile_cat: